src/library/scala/math/Ordering.scala

```diff
@@ -103,10 +104,10 @@ trait Ordering[T] extends Comparator[T] with PartialOrdering[T] with Serializabl
   override def equiv(x: T, y: T): Boolean = compare(x, y) == 0
 
   /** Return `x` if `x` >= `y`, otherwise `y`. */
-  def max(x: T, y: T): T = if (gteq(x, y)) x else y
+  @uncheckedOverride def max[U <: T](x: U, y: U): U = if (gteq(x, y)) x else y
```

Comment:
Seeing this, I just realized we have another problem. This signature change breaks TASTy callers. TASTy signatures contain the number of type arguments, in addition to the erased param/result types.

I'm not sure right now how to solve this. :s